### PR TITLE
fix guilt

### LINKED ISSUE #68114: Guilty about Howling Waif
### Describe the bug

So I've noticed that whenever I kill a howling waif my character feels guilty. But my character has already gotten over killing zombie children. I'm just wondering if that's supposed to happen or if that's a bug. Tested it with another character and found that the same thing was happening there.

### Attach save file

N/A

### Steps to reproduce

1. Kill a lot of zombie children
2. Get over it
3. Kill Howling Waif
4. Cry

### Expected behavior

I imagine that since a howling waif is just another child zombie I should probably feel the same emotions when killing it as any other zombie child.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22621.2134 (22H2)
- Game Version: 38e387c [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Bionic Slots [cbm_slots]
]

### Additional context

Not sure if this applies to any other child zombie types.

### PR DESCRIPTION
**Pull request by KeremBabaG**
#### Summary
Bugfixes "fixes guilt spell to use flags"

#### Purpose of change
Kills were counted per id for guilt attack, so this resulted in stuff like killing over 100 zombie children, stopping to feel guilty about it, but still feeling bad about howling waif which are no different. This fixes that.

closes #68114

#### Describe the solution
Count the monsters with a specific flag instead of monster ids. For this I created four new flags for animals, children, humans, and others that do not fit into this category(like blood sacrifice). For monsters without these flags, their number of kills are counted instead.

#### Describe alternatives you've considered
Jsonify the guilt attack, but for now that's beyond me, neither do I have the time for it after school stuff

#### Testing
I made a new character, picked a tempered zweihander, spawned 100 zombie kids (50 zombie children, 25 howling waifs, 25 shrieklings to be precise), and started slashing like a madman. Here are the results:
Ekran görüntüsü 2024-01-14 025253
Ekran görüntüsü 2024-01-14 025221
It works

#### Additional context
lol

### CODE REVIEW

**Comment by KeremBabaG** [comment]
I'm lying on the bed rn and I just realised that I can do what the original said and just make a new json property "guilt_category" as well. Damn

It'll take some time but lemme know if that's a better way to go and I'll change the pr accordingly

**Comment by KeremBabaG** [comment]
didn't touch MoM, Aftershock and Dinomod because they already have the adequate factions

**Comment by RenechCDDA** [comment]
Why use a faction? This is a fragile approach and prone to breakage if factions are introduced, relationships change, or other factors come into play. Use a flag.

**Comment by KeremBabaG** [comment]
> Why use a faction? This is a fragile approach and prone to breakage if factions are introduced, relationships change, or other factors come into play. Use a flag.

hmm let me try that

**Comment by KeremBabaG** [comment]
@RenechCDDA how's this look?

**Comment by RenechCDDA** [comment]
Yeah this approach is looking pretty good. Is `in_mfaction` leftover code? It doesn't seem to be used anywhere.

**Comment by KeremBabaG** [comment]
> Yeah this approach is looking pretty good. Simple, robust, easily extendable if we need to and without potential for unwanted interactions.
> 
> 
> 
> Is `in_mfaction` leftover code? It doesn't seem to be used anywhere.

Yeah from the previous way. I'll remove it

File: src/kill_tracker.cpp [hunk 1]
```diff
@@ -44,6 +44,41 @@ int kill_tracker::kill_count( const species_id &spec ) const
     return result;
 }
 
+int kill_tracker::guilt_kill_count( const mtype_id &mon ) const
+{
+    int animals = 0;
+    int children = 0;
+    int humans = 0;
+    int others = 0;
+    for( auto &it : kills ) {
```
Comment (Maleclypse) [inline src/kill_tracker.cpp:53]: you have a clang error here.

**Pull request merged**